src/components/group.ts

```diff
@@ -52,12 +53,55 @@ export module Components {
     }
 
     public fixedWidth(): boolean {
-      return this.components().every((c) => c.fixedWidth());
+      return this._components().every((c) => c.fixedWidth());
     }
 
     public fixedHeight(): boolean {
-      return this.components().every((c) => c.fixedHeight());
+      return this._components().every((c) => c.fixedHeight());
     }
+
+    /**
+     * @return {Component[]} The Components in this Group.
+     */
+    public components(): Component[] {
+      return this._componentList.slice();
+    }
+
+    protected _components(): Component[] {
+      return this._componentList;
+    }
+
+    /**
+     * Adds a Component to the Group.
+     * 
+     * @param {Component} component
+     * @param {boolean} prepend If true, prepends the Component. If false, appends it.
+     */
+    public add(component: Component, prepend = false) {
```

Comment:
We should try and think of a better way to handle the boolean flag for prepend/append (ideally not use a boolean flag here at all).